Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -27,9 +27,14 @@ class DataPipelineImplementation: DataPipelineInstance {
         analytics.identify(userId: identifier, traits: body)
     }
 
-    var registeredDeviceToken: String?
+    var registeredDeviceToken: String? {
+        analytics.find(pluginType: DeviceToken.self)?.token
```

Comment:
hmm does it fetch it from segment storage?

Reply:
This feels weird to me for the core CDP SDK to ask an optional plugin for a important piece of data: a device token. 

A device token is a really important piece of data our SDK relies on for a lot of feature. I would expect that the CDP SDK (or segment) keeps track of this piece of data and the device token plugin itself refers to the CDP/Segment SDK. Instead of the other way around, as shown here.

Reply:
I think token isn't really important for segment because it is mainly for tracking. They just have [this small extension](https://github.com/customerio/cdp-analytics-swift/blob/main/Sources/Segment/Plugins/DeviceToken.swift#L33) to add token. But I can't see anywhere segment storing it. I'll be making another PR today to store/fetch token and utilize it, that might help make things more clear.